torchgeo/datasets/geo.py

```diff
@@ -375,6 +377,17 @@ def __init__(
                 f"No {self.__class__.__name__} data was found in '{root}'"
             )
 
+        if bands and self.all_bands:
+            band_indexes = [self.all_bands.index(i) + 1 for i in bands]
+            self.bands = bands
+            assert len(band_indexes) == len(self.bands)
+        elif bands:
+            raise AssertionError("all_bands must be specified")
```

Comment:
This error message isn't really clear about what the problem or solution are. How about something like:
```python
msg = f"{self.__class__.__name__} is missing an `all_bands` attribute, so `bands` cannot be specified."
raise AssertionError(msg)
```
May need to split onto two lines due to line length.